src/InterstitialAdManager.js

```diff
@@ -12,11 +12,19 @@ import { NativeModules } from 'react-native';
 
 const { CTKInterstitialAdManager } = NativeModules;
 
+let pendingPromise: ?Promise<boolean> = null;
 export default {
   /**
    * Shows interstitial ad for a given placementId
    */
   showAd(placementId: string): Promise<boolean> {
-    return CTKInterstitialAdManager.showAd(placementId);
```

Comment:
You are right, but I think the perf. differences are minimal. I imagine we could do it here and remove keeping a reference to it entirely on the backend side. I wanted to keep the JS API surface minimal, with exposing only necessary set of flow types.